src/protocol.rs

```diff
@@ -510,66 +510,88 @@ impl<'a> Protocol<'a> {
             Wss(cow) => Wss(Cow::Owned(cow.into_owned())),
         }
     }
+
+    pub fn tag(&self) -> &'static str {
+        use self::Protocol::*;
+        match self {
+            Dccp(_) => "dccp",
+            Dns(_) => "dns",
+            Dns4(_) => "dns4",
+            Dns6(_) => "dns6",
+            Dnsaddr(_) => "dnsaddr",
+            Http => "http",
+            Https => "https",
+            Ip4(_) => "ip4",
+            Ip6(_) => "ip6",
+            P2pWebRtcDirect => "p2p-webrtc-direct",
+            P2pWebRtcStar => "p2p-webrtc-star",
+            WebRTC => "webrtc",
+            Certhash(_) => "certhash",
+            P2pWebSocketStar => "p2p-websocket-star",
+            Memory(_) => "memory",
+            Onion(_, _) => "onion",
+            Onion3(_) => "onion3",
+            P2p(_) => "p2p",
+            P2pCircuit => "p2p-circuit",
+            Quic => "quic",
+            Sctp(_) => "sctp",
+            Tcp(_) => "tcp",
+            Tls => "tls",
+            Noise => "noise",
+            Udp(_) => "udp",
+            Udt => "udt",
+            Unix(_) => "unix",
+            Utp => "utp",
+            Ws(ref s) if s == "/" => "ws",
+            Ws(_) => "x-parity-ws",
+            Wss(ref s) if s == "/" => "wss",
+            Wss(_) => "x-parity-wss",
+        }
+    }
 }
 
 impl<'a> fmt::Display for Protocol<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         use self::Protocol::*;
+        write!(f, "/{}", self.tag())?;
         match self {
-            Dccp(port) => write!(f, "/dccp/{}", port),
-            Dns(s) => write!(f, "/dns/{}", s),
-            Dns4(s) => write!(f, "/dns4/{}", s),
-            Dns6(s) => write!(f, "/dns6/{}", s),
-            Dnsaddr(s) => write!(f, "/dnsaddr/{}", s),
-            Http => f.write_str("/http"),
-            Https => f.write_str("/https"),
-            Ip4(addr) => write!(f, "/ip4/{}", addr),
-            Ip6(addr) => write!(f, "/ip6/{}", addr),
-            P2pWebRtcDirect => f.write_str("/p2p-webrtc-direct"),
-            P2pWebRtcStar => f.write_str("/p2p-webrtc-star"),
-            WebRTC => f.write_str("/webrtc"),
+            Dccp(port) => write!(f, "/{}", port),
+            Dns(s) => write!(f, "/{}", s),
+            Dns4(s) => write!(f, "/{}", s),
+            Dns6(s) => write!(f, "/{}", s),
+            Dnsaddr(s) => write!(f, "/{}", s),
+            Ip4(addr) => write!(f, "/{}", addr),
+            Ip6(addr) => write!(f, "/{}", addr),
             Certhash(hash) => write!(
                 f,
-                "/certhash/{}",
+                "{}",
```

Comment:
```suggestion
                "/{}",
```

See CI failure:

```

failures:

---- construct_success stdout ----
thread 'construct_success' panicked at 'assertion failed: `(left == right)`
  left: `"/ip4/127.0.0.1/udp/1234/webrtc/certhashuEiDDq4_xNyDorZBH3TlGazyJdOWSwvo4PUo5YHFMrvDE8g"`,
 right: `"/ip4/127.0.0.1/udp/1234/webrtc/certhash/uEiDDq4_xNyDorZBH3TlGazyJdOWSwvo4PUo5YHFMrvDE8g"`', tests/lib.rs:158:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

https://github.com/multiformats/rust-multiaddr/actions/runs/3298957291/jobs/5478581940